lib/working_hours/computation.rb

```diff
@@ -183,20 +182,25 @@ def working_time_between from, to, config: nil
         to = in_config_zone(to, config: config)
         distance = 0
         while from < to
+          from_was = from
           # look at working ranges
           time_in_day = from.seconds_since_midnight
           config[:working_hours][from.wday].each do |begins, ends|
             if time_in_day >= begins and time_in_day < ends
-              # take all we can
-              take = [ends - time_in_day, to - from].min
-              # advance time
-              from += take
-              # increase counter
-              distance += take
+              if (to - from) > (ends - time_in_day)
+                # take all the range and continue
+                distance += (ends - time_in_day)
+                from = from.beginning_of_day + ends
```

Comment:
Main difference is here, same reason as in `advance_to_working_time` the precision of `seconds_since_midnight` is not great (ignores nanoseconds) and floating point causes accuracy errors. In this case we know we have to do up to the end of the range so we reset to clean time instead of adding more floating numbers.